pkg/kv/kvserver/logstore/sideload.go

```diff
@@ -73,40 +73,41 @@ type SideloadStorage interface {
 // The provided slice is not modified, though the returned slice may be backed
 // in parts or entirely by the same memory.
 func MaybeSideloadEntries(
-	ctx context.Context, entriesToAppend []raftpb.Entry, sideloaded SideloadStorage,
+	ctx context.Context, input []raftpb.Entry, sideloaded SideloadStorage,
 ) (
 	_ []raftpb.Entry,
 	numSideloaded int,
 	sideloadedEntriesSize int64,
 	otherEntriesSize int64,
 	_ error,
 ) {
-
-	cow := false
-	for i := range entriesToAppend {
-		if !SniffSideloadedRaftCommand(entriesToAppend[i].Data) {
-			otherEntriesSize += int64(len(entriesToAppend[i].Data))
+	var output []raftpb.Entry
+	for i := range input {
+		typ, err := raftlog.EncodingVersion(input[i])
+		if err != nil {
+			return nil, 0, 0, 0, err
+		}
+		if typ != kvserverbase.RaftVersionSideloaded {
+			otherEntriesSize += int64(len(input[i].Data))
 			continue
 		}
+
 		log.Event(ctx, "sideloading command in append")
-		if !cow {
-			// Avoid mutating the passed-in entries directly. The caller
-			// wants them to remain "fat".
-			log.Eventf(ctx, "copying entries slice of length %d", len(entriesToAppend))
-			cow = true
-			entriesToAppend = append([]raftpb.Entry(nil), entriesToAppend...)
+		if output == nil {
+			// We're seeing the first command that we will sideload, so populate the
+			// output slice with a copy of the input, so that we can replace
+			// individual entries.
+			log.Eventf(ctx, "copying entries slice of length %d", len(input))
+			output = append([]raftpb.Entry(nil), input...)
 		}
-
-		ent := &entriesToAppend[i]
-		cmdID, data := kvserverbase.DecodeRaftCommand(ent.Data) // cheap
+		outputEnt := &output[i]
```

Comment:
nit: s/outputEnt/ent/

Reply:
I think it's worth being explicit here. Note how we have an input slice (which we promised not to mutate) and an output slice. I actually introduced a bug here during development (which led to mutating the input slice) and so I clarified it and am inclined to leave as is.